src/FixerNameValidator.php

```diff
@@ -27,6 +27,7 @@ public function isValid(string $name, bool $isCustom): bool
             return 1 === Preg::match('/^[a-z][a-z0-9_]*$/', $name);
         }
 
-        return 1 === Preg::match('/^[A-Z][a-zA-Z0-9]*\/[a-z][a-z0-9_]*$/', $name);
+        // See: https://regex101.com/r/UngJUn/4
+        return 1 === Preg::match('/^[a-zA-Z][a-zA-Z0-9]*(?!$)((\/[a-z][a-z0-9_]*)?|((?i)(?<!PhpCsFixer)(?-i)\\\\[a-zA-Z][a-zA-Z0-9_]*)*){1}$/', $name);
```

Comment:
the regex seems way too complex

`a-zA-Z` and `(?i)` can be replaced with `i` global modifier

`{1}` is useless, one repetition is by default ;-)

Reply:
[v4](https://regex101.com/r/UngJUn/4): 6 matches (1 002 steps, 0.5ms)
[v5](https://regex101.com/r/UngJUn/5): 7 matches (827 steps, 0.1ms)

Quicker, but `i` modifier allows `VENDOR/SHOUT` which previously (in my regex, but also in original one) was rejected (part after `/`).

Reply:
FQCN should validate againts `TypeExpression::REGEX_IDENTIFIER` regex introduced in #7010

Reply:
I don't know if it's required to check for class existence at this point - we only need to know if it's `vendor/name` (current custom rule naming convention) or FQCN-like string. If rule's name used in the config (like `Foo::class => true`) is invalid, then it will fail anyway.

Reply:
btw, `[Foo::getName() => true]`

Reply:
I do not see big benefit of using `Foo::class` over `Foo::getName()`

Reply:
@keradus `FixerInterface::getName()` is not static, so it should be `(new FooFixer())->getName()` which does not make sense.